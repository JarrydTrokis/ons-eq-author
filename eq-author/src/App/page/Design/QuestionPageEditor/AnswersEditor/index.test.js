import React from "react";
import { shallow } from "enzyme";

import { AnswersEditor } from "./";

describe("Answers Editor", () => {
  jest.useFakeTimers();

  let props;

  beforeEach(() => {
    props = {
      answers: [{ id: "1" }, { id: "2" }],
      onUpdate: jest.fn(),
      onAddOption: jest.fn(),
      onUpdateOption: jest.fn(),
      onDeleteOption: jest.fn(),
      onAddExclusive: jest.fn(),
      onDeleteAnswer: jest.fn(),
      moveAnswer: jest.fn(),
    };
  });

  it("should render", () => {
    const wrapper = shallow(<AnswersEditor {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render the collapsible editor and additional props for the collapsible given", () => {
    const extraProps = {
      canMoveDown: false,
      canMoveUp: false,
      onMoveUp: jest.fn(),
      onMoveDown: jest.fn(),
    };
    const wrapper = shallow(<AnswersEditor {...props} />).renderProp(
      "children"
    )(extraProps, props.answers[0]);

    expect(wrapper.props()).toMatchObject({
      answer: props.answers[0],
      ...extraProps,
    });
  });
});
