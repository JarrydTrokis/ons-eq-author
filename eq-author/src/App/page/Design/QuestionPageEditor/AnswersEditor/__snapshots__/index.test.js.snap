// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Answers Editor should render 1`] = `
<Reorder
  Transition={[Function]}
  list={
    Array [
      Object {
        "id": "1",
      },
      Object {
        "id": "2",
      },
    ]
  }
  onMove={[MockFunction]}
>
  <Component />
</Reorder>
`;
